### PR TITLE
Vcu validators

### PR DESCRIPTION
**Pull request by BradHacker**

### CODE REVIEW

**Comment by BradHacker** [comment]
@1nv8rzim @P4ral1ax we need to validate all of the validation types. See [c653ea1](https://github.com/BradHacker/laforge/pull/3/commits/c653ea119cadfd6765e4f8ec4778514684eb1828) for an example of logic errors in the validation code.

File: grpc/agent/agent_unix.go [hunk 1]
```diff
@@ -192,3 +167,320 @@ func GetSystemDependencies() []string {
 		"Requires=network.target",
 		"After=network-online.target"}
 }
+
+// Validation functions
+func GetNetBanner(portnum int64) (bool, error) { // exists (boolean)
+	return true, nil
+}
+
+func NetHttpContentRegex(full_url string) (string, error) { // content hash (string)
```
Comment (P4ral1ax) [inline grpc/agent/agent_unix.go:176]: This function doesn't do any regex?

File: grpc/agent/agent.go [hunk 2]
```diff
@@ -274,6 +275,93 @@ func RequestTask(c pb.LaforgeClient) {
...
+			case "net-udp-open":
+				ip := taskArgs[1]
+				port, err := strconv.Atoi(taskArgs[2])
+				if err != nil {
+					RequestTaskStatusRequest(strconv.FormatBool(false), err, r.Id, c)
+				}
+				open, err := NetUDPOpen(ip, port, "udp-test")
+				RequestTaskStatusRequest(strconv.FormatBool(open), err, r.Id, c)
+			case "net-http-content-regex":
+				ip := taskArgs[1]
```
Comment (P4ral1ax) [inline grpc/agent/agent.go]: Shouldn't be "ip" for a url

File: grpc/agent/agent_unix.go [hunk 3]
```diff
@@ -192,3 +167,320 @@ func GetSystemDependencies() []string {
...
+	}
+	defer net_resp.Body.Close()
+	page_html, deserialize_err := ioutil.ReadAll(net_resp.Body)
+	if deserialize_err != nil {
+		return "", deserialize_err
+	}
+
+	// return MD5Sum(fmt.Sprintf("%s", page_html)), nil
+	return string(page_html[:]), nil // stringify
+}
```
Comment (P4ral1ax) [inline grpc/agent/agent_unix.go]: This function doesn't do any regex anything. It just gets the content and returns it. The agent.go function doesn't do regex either?

**Comment by BradHacker** [comment]
Everything checks out and all test validations are now passing

**Review by 1nv8rzim — APPROVED** [review]

**Review by P4ral1ax — APPROVED** [review]

**Pull request merged**